scripts/migration/v0.0.4__add-room-templates.ts

```diff
@@ -0,0 +1,119 @@
+import { MigrateOptions } from "fireway";
+import { isEqual } from "lodash";
+
+import { getVenueId } from "../../functions/src/utils/venue";
+import { Room } from "../../src/types/rooms";
+
+const isVenueRegex = /\/in\/(\w+)$/;
+
+interface CreateVenueFromPortalProps {
+  firestore: FirebaseFirestore.Firestore;
+  portal: Room;
+  parentVenueId: string;
+  parentWorldId: string;
+}
+
+const createVenueFromPortal = async ({
+  firestore,
+  portal,
+  parentVenueId,
+  parentWorldId,
+}: CreateVenueFromPortalProps) => {
+  const venueData = {
+    name: portal.title,
+    template: "zoomroom",
+    zoomUrl: portal.url ?? "",
+    parentId: parentVenueId,
+    worldId: parentWorldId,
+  };
+
+  let venueExists = true;
+  let index = -1;
+  const baseVenueId = getVenueId(portal.title);
+  let venueId = baseVenueId;
+  while (venueExists) {
+    index += 1;
+    venueId =
+      (baseVenueId || "externalexperience") + (index === 0 ? "" : index);
+    const venue = await firestore.collection("venues").doc(venueId).get();
+    venueExists = venue.exists;
+  }
+
+  await firestore.collection("venues").doc(venueId).set(venueData);
+
+  return `/in/${venueId}`;
+};
+
+export const migrate = async ({ firestore }: MigrateOptions) => {
+  const { docs: venueDocs } = await firestore
+    .collection("venues")
+    .where("template", "==", "partymap")
+    .get();
+
+  await Promise.all(
```

Comment:
I think we're going to need some abstractions to help make this kind of iteration in a migration much easier. Not expecting you to do this now. Worth thinking about for next time.